Sources/GRPC/CallHandlers/ServerStreamingCallHandler.swift

```diff
@@ -83,7 +83,7 @@ public final class ServerStreamingCallHandler<
     }
   }
 
-  override internal func sendErrorStatus(_ status: GRPCStatus) {
-    self.callContext?.statusPromise.fail(status)
+  internal override func sendErrorStatusAndMetadata(_ statusAndMetadata: GRPCStatusAndMetadata) {
+    self.callContext?.statusPromise.fail(statusAndMetadata.status)
```

Comment:
I think we need to add the metadata to the call context's trailing metadata here